ruby/tzdump.rb

```diff
@@ -5,35 +7,50 @@
 START_UTC = DateTime.new(1905, 1, 1, 0, 0, 0, '+0')
 END_UTC = DateTime.new(2035, 1, 1, 0, 0, 0, '+0')
 
+def puts_crlf(*messages)
+  $stdout.write(messages.join("\r\n"))
+  $stdout.write("\r\n")
+end
+
 def format_offset(offset)
   sign = offset < 0 ? "-" : "+"
   offset = offset.abs
   return '%s%02d:%02d:%02d' % [sign, offset / 3600, (offset / 60) % 60, offset % 60]
 end
 
-def dump_zone(console, id)
-  console.write "#{id}\r\n"
+def dump_period(period, zone)
```

Comment:
In the conversion to classes, it gets easy to add an iterator using `to_enum()`.
We can probably convert this to a function + yield to a block however.